src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -0,0 +1,122 @@
+            {
+                return;
+            }
+
+            var method = syntaxContext.TargetToken.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            if (method is not null && !method.GetModifiers().Any(SyntaxKind.AsyncKeyword))
+            {
+                var symbol = semanticModel.GetDeclaredSymbol(method, cancellationToken) as IMethodSymbol;
+                if (symbol is not null && IsTask(symbol.ReturnType))
+                {
+                    _shouldMakeContainerAsync = true;
+                    context.AddItem(GetCompletionItem());
+                    return;
+                }
+            }
+
+            context.AddItem(GetCompletionItem());
+
+            // Local functions.
+            static bool IsTask(ITypeSymbol returnType)
+                => returnType.Name is "Task" or "ValueTask";
+
+            CompletionItem GetCompletionItem() => CommonCompletionItem.Create(
+                displayText: SyntaxFacts.GetText(SyntaxKind.AwaitKeyword),
+                displayTextSuffix: "",
+                rules: CompletionItemRules.Default,
+                Glyph.Keyword,
+                description: RecommendedKeyword.CreateDisplayParts(SyntaxFacts.GetText(SyntaxKind.AwaitKeyword), string.Empty),
+                inlineDescription: _shouldMakeContainerAsync ? CSharpFeaturesResources.Make_container_async : null,
+                isComplexTextEdit: _shouldMakeContainerAsync);
+        }
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem completionItem, TextSpan completionListSpan, char? commitKey, bool disallowAddingImports, CancellationToken cancellationToken)
+        {
+            if (!_shouldMakeContainerAsync)
+            {
+                return await base.GetChangeAsync(document, completionItem, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var declaration = root?.FindToken(completionListSpan.Start).GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            if (root is null || declaration is null)
+            {
+                return await base.GetChangeAsync(document, completionItem, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+            }
+
+            // MethodDeclarationSyntax, LocalFunctionStatementSyntax, AnonymousMethodExpressionSyntax, ParenthesizedLambdaExpressionSyntax, SimpleLambdaExpressionSyntax.
+
+            var documentWithAsyncModifier = document.WithSyntaxRoot(root.ReplaceNode(declaration, AddAsyncModifier(document, declaration)));
+            var formattedDocument = await Formatter.FormatAsync(documentWithAsyncModifier, Formatter.Annotation, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
This looks *very* unneeded. But removing it causes `async` to not have a space after it.

I already have `WithAdditionalAnnotations(Formatter.Annotation)` on the declaration node. Not sure why this doesn't correctly format it?

@genlu Any idea?

Reply:
I might be wrong, but I don't think completion would automatically format the text (like what code actions do), so this is required.

Reply:
i think using SyntaxGenerator/Formatting here is *very* overkill.  We're adding `async ` to the method.  Just go find the method and literally just add that token with the right trivia on it :)